src/XMakeBuildEngine/Microsoft.Build.csproj

```diff
@@ -13,10 +13,10 @@
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <DefineConstants>TRACE;DEBUG;STANDALONEBUILD;BUILD_ENGINE</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;STANDALONEBUILD;BUILD_ENGINE;$(DefineConstantsMsbuildDebugger)</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DefineConstants>TRACE;STANDALONEBUILD;BUILD_ENGINE</DefineConstants>
+    <DefineConstants>TRACE;STANDALONEBUILD;BUILD_ENGINE;$(DefineConstantsMsbuildDebugger)</DefineConstants>
```

Comment:
We shouldn't need to set the constants here, they should come from the dirs.props. If BUILD_ENGINE needs to be in here it can be added to the list.